### ISSUE TITLE
Update problem - Failed update while auto update is turned off

### ISSUE THREAD

**Issue by Poonsta** [safing/portmaster#1383]
**Pre-Submit Checklist**:

- Check applicable sources for existing issues:
  - [Windows Known Issues](https://docs.safing.io/portmaster/install/windows#known-issues)
  - [Linux Known Issues](https://docs.safing.io/portmaster/install/linux#compatibility)
  - [Github Issues](https://github.com/safing/portmaster/issues?q=is%3Aissue+label%3Abug)

**What happened**:
Portmaster notifier did not start. I could not start the notifier and the UI at all. Restarting the computer did not help. 
I noticed that the Portmaster core in Task Manager showed a new name: portmaster-core_v1-6-2.exe
Looks like Portmaster did a failed update, even though I disabled auto update in general settings.
I downloaded the new version of Portmaster, and tried to install it. The installation started but failed at downloading the 300MB part. I tried installing it a few times, same result each time.
I uninstalled Portmaster completely, removed every remnant of it from the system manually. This time the install went through, notifier starts automatically, and can start the UI.

**What did you expect to happen?**:
No update until I check manually, and when I do, it updates with no problem.

**How did you reproduce it?**:
I did not reproduce it, I don't know how to.

**Debug Information**:

It is from the newly installed version, not sure how much help it is.

<details>
<summary>Version 1.6.2</summary>

```

Portmaster
version 1.6.2

commit tags/v1.6.2-0-ge6aba884ae7e4cd49cc02d150807cfd016338073
built with go1.21.2 (gc) windows/amd64
  using options 
  by user@docker
  on 01.12.2023

Licensed under the AGPLv3 license.
The source code is available here: https://github.com/safing/portmaster
```

</details>

<details>
<summary>Platform: Microsoft Windows 10 Pro 10.0.19045 Build 19045</summary>

```
System: Microsoft Windows 10 Pro windows (Standalone Workstation) 10.0.19045 Build 19045
Kernel: 10.0.19045 Build 19045 x86_64

```

</details>

<details>
<summary>No Module Error</summary>

</details>

<details>
<summary>Unexpected Logs</summary>

```
231211 15:13:45.719 tel/entity:253 > WARN 002 intel: failed to get location data for 2606:4700:4700::1111: geoip database not available
231211 15:13:50.192 olver-mdns:114 > WARN 004 intel(mdns): failed to create udp6 listen multicast socket: listen udp6 [ff02::fb]:5353: setsockopt: not supported by windows
231211 15:55:00.176 CURRENT TIME
```

</details>

<details>
<summary>Status: Online</summary>

```
OnlineStatus:          Online
CaptivePortal:         
```

</details>

<details>
<summary>SPN: disabled (module disabled)</summary>

```
HomeHubID:    
HomeHubName:  
HomeHubIP:    
Transport:    
---
Client:       true
PublicHub:    false
HubHasIPv4:   false
HubHasIPv6:   false
```

</details>

<details>
<summary>Resolvers: 8/8</summary>

```
Cloudflare (dot://cloudflare-dns.com:853#config)
  dot://cloudflare-dns.com:853#config
  Failing: false

Cloudflare (dot://cloudflare-dns.com:853#config)
  dot://cloudflare-dns.com:853#config
  Failing: false

Cloudflare (dot://cloudflare-dns.com:853#config)
  dot://cloudflare-dns.com:853#config
  Failing: false

Cloudflare (dot://cloudflare-dns.com:853#config)
  dot://cloudflare-dns.com:853#config
  Failing: false

2606:4700:4700::1111 (dns://2606:4700:4700::1111:53#system)
  dns://2606:4700:4700::1111:53#system
  Failing: false

2620:119:35::35 (dns://2620:119:35::35:53#system)
  dns://2620:119:35::35:53#system
  Failing: false

1.1.1.1 (dns://1.1.1.1:53#system)
  dns://1.1.1.1:53#system
  Failing: false

208.67.222.222 (dns://208.67.222.222:53#system)
  dns://208.67.222.222:53#system
  Failing: false
```

</details>

<details>
<summary>Config: 9</summary>

```
core/automaticUpdates: false
core/expertiseLevel: expert
core/metrics/instance: [redacted]
dns/nameservers: [redacted]
dns/noMulticastDNS: true
filter/defaultAction: ask
filter/endpoints: [redacted]
filter/lists: [TRAC MAL BAD WSP UNBREAK]
spn/enable: false
```

</details>

<details>
<summary>Updates: stable (13/31)</summary>

```
Active:
all/intel/geoip/geoipv4.mmdb.gz: 20231201.0.1
all/intel/geoip/geoipv6.mmdb.gz: 20231201.0.1
all/intel/lists/base.dsdl: 20231201.0.7
all/intel/lists/index.dsd: 2023.6.13
all/intel/lists/intermediate.dsdl: 20231210.0.4
all/intel/lists/urgent.dsdl: 20231211.9.42
all/intel/portmaster/news.yaml: 20231201.10.3
all/intel/portmaster/notifications.yaml: 20231211.13.25
all/ui/modules/assets.zip: 0.3.1
all/ui/modules/portmaster.zip: 0.8.2
windows_amd64/core/portmaster-core.exe: 1.6.2
windows_amd64/kext/portmaster-kext.sys: 1.1.2
windows_amd64/start/portmaster-start.exe: 1.0.13

Selected:
all/dashboards/hubs.json: 0.0.2
all/intel/geoip/geoipv4-mini.mmdb.gz: 20231201.0.1
all/intel/geoip/geoipv4.mmdb.gz: 20231201.0.1
all/intel/geoip/geoipv6-mini.mmdb.gz: 20231201.0.1
all/intel/geoip/geoipv6.mmdb.gz: 20231201.0.1
all/intel/lists/base.dsdl: 20231201.0.7
all/intel/lists/index.dsd: 2023.6.13
all/intel/lists/intermediate.dsdl: 20231210.0.4
all/intel/lists/urgent.dsdl: 20231211.9.42
all/intel/portmaster/news.yaml: 20231201.10.3
all/intel/portmaster/notifications.yaml: 20231211.13.25
all/intel/spn/main-intel.yaml: 20231103.10.14
all/ui/modules/assets.zip: 0.3.1
all/ui/modules/base.zip: 0.2.11
all/ui/modules/console.zip: 0.1.11
all/ui/modules/monitor.zip: 0.2.4
all/ui/modules/portmaster.zip: 0.8.2
all/ui/modules/profilemgr.zip: 0.1.7
all/ui/modules/settings.zip: 0.1.8
windows_amd64/app/portmaster-app.zip: 0.2.6
windows_amd64/core/portmaster-core.exe: 1.6.2
windows_amd64/hub/spn-hub.exe: 0.7.3
windows_amd64/jess/jess.exe: 0.3.2
windows_amd64/kext/portmaster-kext.dll: 1.0.14
windows_amd64/kext/portmaster-kext.pdb: 1.1.2
windows_amd64/kext/portmaster-kext.sys: 1.1.2
windows_amd64/notifier/portmaster-notifier.exe: 0.3.6
windows_amd64/notifier/portmaster-snoretoast.exe: 0.6.0
windows_amd64/notifier/portmaster-wintoast.dll: 0.1.4
windows_amd64/packages/portmaster-installer.exe: 1.0.13
windows_amd64/start/portmaster-start.exe: 1.0.13
```

</details>

**Comment by Terrance** [comment]
I think I also hit this (neither the tray icon nor the UI would show, despite any attempts to restart the core service or the UI, or a reboot), and similarly had to reinstall to get things working again.  Unfortunately I did that before finding this issue so I also can't poke the old version any further.

Looking in the logs folder, it does appear to have jumped from 1.5.1 to 1.6.2 with automatic updates disabled.

**Comment by dhaavi** [comment]
I checked and the bug in the auto-updating was already fixed in v1.5.x.
So, I am not sure what happened here.

Please note that Portmaster has two settings for auto updates. One is for software, the other for intelligence data.
Also note that changing the release channel also forces an update to get the right version for the selected channel.

As long as one update option is enabled, Portmaster will automatically check for new versions, but will only download the data enabled for automatic updates.

Please monitor the situation and report back how things go with the next version.

Someone from the community started work on an offline installer, but I don't know when we will have time to integrate that into our workflow. But, things are moving and we'll get there eventually.

The broken updates could stem from our update servers being overloaded and downloads being really slow. Failed downloads however, should not be a problem, because they would never pass the signature check.

**Comment by Poonsta** [comment]
Hey @dhaavi 

Same thing happened again, Portmaster updated to the new version (1.6.5) automatically, even though auto update is off.
Seems like the update is not complete though, I cannot open the app, just like before. The only thing I can do from the notifier (its context menu) is to turn off Portmaster, but that does not turn off the Portmaster service, only the 4 portmaster processes. After turning Portmaster off, I cannot start it again, have to restart Windows. Even after the restart I cannot open the portmaster app.

**Comment by Terrance** [comment]
My installation is similarly broken again; same state as I reported above.

DNS itself seems to be working normally, so I've left it as-is in case there's something useful I can provide for investigation, though no core logs have been generated today so I can't actually tell what version I have installed now (on last boot it was 1.6.2).

**Comment by Terrance** [comment]
For what it's worth, I've left it alone since my previous comment, but the UI still remains unopenable.  Connection prompts continue to show and seemingly work (but I can't check for the newly-added rule without the UI).

> It would be interesting to understand what the issue with those couple machines is.

If you're waiting for us to provide something, you'll need to tell me what more I can give you.

So far I've found this output when trying to launch the UI manually:

```
PS C:\ProgramData\Safing\Portmaster> .\portmaster-start.exe app --data=C:\ProgramData\Safing\Portmaster
[pmstart] 2024/02/04 17:51:05 WARNING: portmaster-start is marked as a GUI application in order to get rid of the console window.
[pmstart] 2024/02/04 17:51:05 WARNING: CTRL-C will immediately kill without clean shutdown.
[pmstart] 2024/02/04 17:51:05 app/portmaster-app.zip failed with: could not get component: the requested file is not available locally
Error: could not get component: the requested file is not available locally
```

**Comment by Terrance** [comment]
That got things working last time and presumably will this time too, but that just delays the problem until your next update ignores auto-update and breaks it again.

It sounds like you've not been able to reproduce the problem on your end so far?

**Comment by Terrance** [comment]
I needed to get back in to adjust some rules, so I went ahead and reinstalled again.  That got things working at the time, but now (a few hours later) as I try to get back in again, I note that `portmaster-start.exe` was renamed at some point later on to `portmaster-start.exe.bak`, meaning the Start Menu shortcut and notifier can't launch it any more.  Renaming it back and attempting to launch the UI, I get an unclosable window with the following error:

> open C:\ProgramData\Safing\Portmaster\updates\all\ui\modules\portmaster_v0-7-12.zip: The system cannot find the file specified.

A package for 0.7.12 indeed does not exist (present are 0.7.13 and 0.8.2 dated last year, and 0.8.3 added today at the reinstall time), so there seems to be some sort of version mismatch going on here -- I'm guessing this is down to a combination of the reinstall not updating the components (due to auto-update disabled or otherwise) and the old packages being tidied up somehow, despite still being used?

(I've installed via Chocolatey, and triggered a reinstall using `choco install --force portmaster` which forces a re-run of [portmaster-installer_v1-0-0.exe](https://updates.safing.io/windows_amd64/packages/portmaster-installer_v1-0-0.exe).)

**Comment by Raphty** [comment]
you are talkig about this/ https://community.chocolatey.org/packages/portmaster

we did not submit this, nor do we maintain this. pleas use the official install methods.

**Comment by Terrance** [comment]
Noted, though I've downloaded and ran `portmaster-installer.exe` (identifies as 1.0.13.0) from your home page and the mismatching UI package persists.  The state remains as described above with the main app not launchable -- seems like it's not updating any of the existing components, and `portmaster-start.exe` is trying to use a hard-coded version that doesn't exist?

**Comment by Raphty** [comment]
I seems this issue has evolved away from the original topic of updating while auto update is turned off.

I will close the issue for now, if there is more to add regarding the auto update situation pleas comment and I will reopen it.

@Terrance your sub issue seems to be a configuration issue, that I have not seen before, pleas ask the community for help with your specific setup.

**Comment by shaund** [comment]
I have what sounds like a similar issue. Auto-updates disabled, on Windows 10 Enterprise, official exe installer.

If I minimize the main portmaster window, I can't open it from the taskbar icon (right clicking doesn't pop up a menu, double-clicking does nothing). It's still running and connection notifications continue, but no matter what I do I can't get the main portmaster window to open. If I kill `portmaster-core_v1-6-5.exe`, `portmaster-notifier_v0-3-6.exe`, and the two `portmaster-start.exe` (one with the portmaster core service under it), then try to restart Portmaster, nothing works and I have to restart the entire computer.

I've tried uninstalling and reinstalling, but the problem persists.

I've had this problem since I first installed portmaster back in Feb 2023, not sure what version that was.

**Comment by Terrance** [comment]
The notifier app's tray icon becoming unresponsive is a long-standing issue (see also #1105) and probably unrelated to this, unless opening Portmaster from the Start menu also doesn't work.

If that is the case, try [launching the UI manually from a console](https://github.com/safing/portmaster/issues/1383#issuecomment-1925864549) and see if you get a similar error message (NB. This was before the `portmaster-start.exe` version mismatch woes reported later in here).

**Comment by Terrance** [comment]
I had been holding off on updating since 1.6.5, and on my end it has correctly refrained from automatically updating when automatic updates are disabled, so hopefully that option is now working as intended.

I did however get a repeat of https://github.com/safing/portmaster/issues/1383#issuecomment-1937895628 which I've moved to a separate issue #1563.
